.buildkite/pipeline.yml

```diff
@@ -39,27 +23,34 @@ steps:
 
           echo "--- :rspec: Run Rspec"
           bundle exec rspec --profile 10 --format progress
-        env: *xcode_image
-        plugins: [*ci_toolkit]
+
+          echo "--- :fastlane: :buildkite: Test pipeline upload action"
+          bundle exec fastlane run buildkite_pipeline_upload pipeline_file:.buildkite/pipeline_for_upload_action_test.yml
+        env:
+          IMAGE_ID: $IMAGE_ID
+        plugins:
+          - $CI_TOOLKIT_PLUGIN
         agents:
-          queue: "mac"
+          queue: mac
         matrix:
           setup:
             ruby:
-              - 3.2.2
+              - $RUBY_VERSION
 
-  #################
-  # Push to RubyGems
-  #################
   - label: ":rubygems: Publish to RubyGems"
-    key: "gem-push"
+    key: gem-push
     if: build.tag != null
     depends_on:
      - test
     # Note: We intentionally call a separate `.sh` script here (as opposed to having all the
     # commands written inline) to avoid leaking a key used in the process in clear in the
     # BUILDKITE_COMMAND environment variable.
     command: .buildkite/commands/gem-push.sh
-    plugins: [*docker_plugin]
+    plugins:
+      - docker#v5.8.0:
```

Comment:
This version could be moved to the shared vars, too, but given it's used only once I haven't done it yet.

Conversely, the CI toolkit was already setup so I leaned into it rather than removing it.